### DISCUSSION TITLE
ADBSocket question...and then maybe more

### DISCUSSION THREAD

**Discussion by rickx** [yungd1plomat/AdvancedSharpAdbClient#114] [Ideas]
Hi guys,

thanks for this. I've been using this library's predecessor and this new development looks nice. Before proposing something, I have this question: to use monitoring (DeviceMonitor), one single socket must be used all the time. Is that correct?
Because, if - like I guess - it is, then the whole approach of this library make things very difficult if you want to use it to interact with multiple devices over a single work session (i.e. within the app's lifetime). So, before going further, can you answer this? Maybe I've missed something.

**Comment by wherewhere** [comment]
DeviceMonitor is monitor the device list of connected devices, and it is provided by adb server. The socket disconnect only when you disconnected it or adb is exited. So you just need one monitor per process to monitoring the device list. If you have multiple threads, you can create a static monitor and use reference counter to manage it.
